giraffe/src/transforms/band.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  BandLayerSpec,
+  Band,
+  ColumnGroupMap,
+  CumulativeValuesByTime,
+  LineData,
+  LinePosition,
+  NumericColumnData,
+  Table,
+  BandIndexMap,
+} from '../types'
+import {FILL, VALUE, RESULT, MAX, MIN} from '../constants/columnKeys'
+import {createGroupIDColumn, getNominalColorScale} from './'
+
+export const getBands = (
+  fill: ColumnGroupMap,
+  lineData: LineData,
+  bandFillColors: string[]
+): Band[] => {
+  const minMaxMap = {}
+  const {columnKeys, mappings} = fill
+  const columnKeysWithoutResult = Array.isArray(columnKeys)
+    ? columnKeys.filter(key => key !== RESULT)
+    : []
+
+  const bands = Array.isArray(mappings)
+    ? mappings.reduce((bandLines, map, index) => {
+        const lineName = columnKeysWithoutResult.reduce((accum, current) => {
+          return `${accum}${mappings[index][current]}`
+        }, '')
+        if (!minMaxMap[lineName]) {
+          minMaxMap[lineName] = {}
+        }
+        if (map[RESULT] === MAX) {
+          if (!minMaxMap[lineName][MAX]) {
+            minMaxMap[lineName][MAX] = lineData[index]
+          }
+        } else if (map[RESULT] === MIN) {
+          if (!minMaxMap[lineName][MIN]) {
+            minMaxMap[lineName][MIN] = lineData[index]
+          }
+        } else {
+          bandLines.push({
+            ...lineData[index],
+            lineName,
+            fill: bandFillColors[bandLines.length],
+          })
+        }
+        return bandLines
+      }, [])
+    : []
+
+  bands.forEach(line => {
+    if (minMaxMap[line.lineName][MIN]) {
+      line[MIN] = {...minMaxMap[line.lineName][MIN], fill: line.fill}
+    }
+    if (minMaxMap[line.lineName][MAX]) {
+      line[MAX] = {...minMaxMap[line.lineName][MAX], fill: line.fill}
+    }
+  })
+  return bands as Band[]
+}
+
+export const getBandIndexMap = (
+  fillColumnMap: ColumnGroupMap
+): BandIndexMap => {
+  const bandIndices = {
+    rowIndices: [],
+    minIndices: [],
+    maxIndices: [],
+  }
+
+  if (Array.isArray(fillColumnMap.mappings)) {
+    fillColumnMap.mappings.forEach((line, index) => {
+      if (line[RESULT] === MAX) {
+        bandIndices.maxIndices.push(index)
+      } else if (line[RESULT] === MIN) {
+        bandIndices.minIndices.push(index)
+      } else {
+        bandIndices.rowIndices.push(index)
+      }
+    })
+  }
+
+  return bandIndices
+}
+
+export const getBandLineIndices = (fillColumnMap: ColumnGroupMap): number[] => {
```

Comment:
I'm not sure if this is because the lines are based off the fillColumn, but when you unselect the fill: cpu knob, all of the lines connect. Is this an expected behavior or an issue?
![ezgif-6-be883319c754](https://user-images.githubusercontent.com/34894628/89931853-cd2f3d00-dbda-11ea-9d48-7bfe34aea2c3.gif)

Reply:
Yes, this is correct in the sense that there is no way Flux will return a CSV without the cpu. This is an issue with the storybook knobs allowing a customization that would never happen. I'll adjust or possibly remove them.